### ISSUE TITLE
dice_score vs multi_class_dice_score -- Mismatch detected

### ISSUE THREAD

**Issue by charleygros** [ivadomed/ivadomed#1091]
Hey team,

Writing to you on behalf of some collaborators. I don't have all the full picture yet, I will post asap (eg config file), when they share with me. But thought I would reach out, in case any of you have already been thinking of this / encountering this issue.

Context:
- 2 class segmentation problem.
- They trained their model, went to the logdir to plot the validation metrics
- `multi_class_dice_score` is mainly < 0.8, while `dice_score` mainly > 0.8

Could be a bunch of things.. maybe class imbalance? `empty_score` param?... Any thoughts?

Will update this thread asap while receiving more info from them.

![multi_class_dice_score](https://user-images.githubusercontent.com/14353425/155920648-57a157ba-9e7c-4061-b800-794b4a42c460.png)
![dice_score](https://user-images.githubusercontent.com/14353425/155920652-4f1a49c8-e639-41f1-837f-39064c88dddd.png)

**Comment by andreanne-lemay** [comment]
Yes, I'd say like you, most probably class imbalance. One class has a lower Dice score than the other but has a relatively low volume. E.g. class 1: 60% Dice score and class 2: 90% Dice score -> Multi-class dice score 75% but if class 2 represents most of the positive class pixels (bigger volume), the Dice score will be much closer to 90%.

**Comment by naga-karthik** [comment]
Slightly tangential but I was looking at the implementations of the `dice_score` and DiceLoss and I noticed that both of them _do include_ the background while computing the Dice score. Consider a 2-class lesion seg problem with the background as class 0 and lesion as class 1. Then, including the background would offset the Dice score to a much higher value, right (because the background will dominate the lesion)? 

I am training models using Monai which has an option of excluding the background and when I am comparing the results with the models trained with ivadomed, there is a huge difference in the DiceLoss and dice_score. Hence the motivation for this comment. I think that we should add a boolean argument to ask whether to include the background or not. Any thoughts?
